### PR TITLE
Change strategy to test connectivity between ODBC driver and SQL plugin

### LINKED ISSUE #57: Add Query Engine Healthcheck for ODBC Driver Connectivity Test
In current code base the ODBC driver retrieves the installed plugins as the method to check the SQL plugin status before establishing the connection. We need to bring up a better/standard way to do the plugin health check for the ODBC driver connectivity test.

Related issue: https://github.com/opendistro-for-elasticsearch/sql/issues/1078

### PR DESCRIPTION
**Pull request by chloe-zh**
### Description
Currently before establishes the connection between the ODBC driver and the OpenSearch server, the client tries to check the installed plugins in the current OpenSearch cluster by calling the `_cat/plugins` API. In this way, the connection will not be established if it does not detect the plugin with exactly the same name that was hard coded in the code base. It might cause compatibility issue once the SQL plugin is renamed, and it does not follow the common strategy to test the odbc connectivity as well. 

Thus we changed the strategy in this PR to let it send a simple `select 1` query to the server to check the sql plugin availability instead.

- Case 1: SQL plugin is not installed
Behavior: connectivity test fails
image

- Case 2: SQL plugin is installed but disabled
Behavior: connectivity test fails
image

- Case 3: SQL plugin is installed and enabled, but somehow unhealthy (query execution failed)
Behavior: connectivity test fails
Similar error windows pop up with message: **Connection error: [OpenSearch][SQL ODBC Driver][SQL Plugin] Connection error: SQL plugin is disabled, please enable the plugin to use this driver.**

- Case 4: SQL plugin is installed, enabled and in healthy status
Behavior: connectivity test passes
image

 
### Issues Resolved
#57 
 
### Check List
- [x] All tests pass, including unit test, integration test and doctest
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: sql-odbc/src/sqlodbc/opensearch_communication.cpp [hunk 1]
```diff
@@ -433,57 +434,71 @@ OpenSearchCommunication::IssueRequest(
...
+        else {
+            std::unique_ptr< OpenSearchResult > result =
+                std::make_unique< OpenSearchResult >();
+            AwsHttpResponseToString(response, result->result_json);
+            std::shared_ptr< ErrorDetails > error_details =
+                ParseErrorResponse(*result);
+
+            if(!IsSQLPluginEnabled(error_details)) {
+                m_error_message +=
+                    "The SQL plugin is disabled. The SQL plugin must be "
```
Comment (penghuo) [inline sql-odbc/src/sqlodbc/opensearch_communication.cpp:475]: could we add the doc link to guide customer how to enable SQL plugin.

File: sql-odbc/src/sqlodbc/opensearch_communication.cpp [hunk 2]
```diff
@@ -433,57 +434,71 @@ OpenSearchCommunication::IssueRequest(
...
+                    "enabled in order to use this driver. Response body: '"
+                    + m_response_str + "'";
+                SetErrorDetails("Connection error", m_error_message,
+                                ConnErrorType::CONN_ERROR_COMM_LINK_FAILURE);
+            }
+
+            if (response->HasClientError()) {
+                m_error_message += " Client error: '"
```
Comment (penghuo) [inline sql-odbc/src/sqlodbc/opensearch_communication.cpp:483]: What is the  action item if the customer receive "Client error" or "Response error". Could we collapsed into single error message.

**Review by penghuo — APPROVED** [review]
Thanks for the change!

**Review by dai-chen — APPROVED** [review]
Thanks for the changes!

**Pull request merged**